encrypted-media/scripts/playback-persistent-license-events.js

```diff
@@ -0,0 +1,168 @@
+                                    + ', set src before setMediaKeys, check events';
+
+    var configuration = {   initDataTypes: [ config.initDataType ],
+                            audioCapabilities: [ { contentType: config.audioType } ],
+                            videoCapabilities: [ { contentType: config.videoType } ],
+                            sessionTypes: [ 'persistent-license' ] };
+
+
+    async_test( function( test ) {
+
+        var _video = config.video,
+            _mediaKeys,
+            _mediaKeySession,
+            _mediaSource,
+            _sessionId,
+            _allKeysUsableEvent = false,
+            _playingEvent = false,
+            _timeupdateEvent = false,
+            _releaseSequence = false,
+            _events = [ ];
+
+        function onMessage(event) {
+            assert_equals( event.target, _mediaKeySession );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            assert_any( assert_equals,
+                        event.messageType,
+                        _releaseSequence    ? [ 'license-release' ]
+                                            : [ 'license-request', 'individualization-request' ] );
+
+            if ( event.messageType !== 'individualization-request' ) {
+                _events.push( event.messageType );
+            }
+
+            config.messagehandler( config.keysystem, event.messageType, event.message ).then( function( response ) {
+                if ( event.messageType === 'license-request' ) {
+                    _events.push( 'license-response' );
+                } else if ( event.messageType === 'license-release' ) {
+                    _events.push( 'release-response' );
+                }
+
+                waitForEventAndRunStep('keystatuseschange', _mediaKeySession, onKeyStatusesChange, test);
+                _mediaKeySession.update( response ).then( function() {
+                    _events.push('updated');
+                }).catch(function(error) {
+                    forceTestFailureFromPromise(test, error);
+                });
+            });
+        }
+
+        function onKeyStatusesChange(event) {
+            assert_equals(event.target, _mediaKeySession );
+            assert_true(event instanceof window.Event );
+            assert_equals(event.type, 'keystatuseschange' );
+
+            var hasKeys = false, pendingKeys = false;
+            _mediaKeySession.keyStatuses.forEach( function( value, keyid ) {
+                assert_any( assert_equals, value, [ 'status-pending', 'usable' ] );
+
+                hasKeys = true;
+                pendingKeys = pendingKeys || ( value === 'status-pending' );
+
+            });
+
+            if ( !_allKeysUsableEvent && hasKeys && !pendingKeys ) {
+                _allKeysUsableEvent = true;
+                _events.push( 'allkeysusable' );
+                _video.setMediaKeys(_mediaKeys);
```

Comment:
No. I was probably trying to make it work on Edge, which seems to require setMediaKeys to be called quite late.

We'll need a more comprehensive test for the timing of setMediaKeys - there seems to be more to it than before / after setting the src.